src/core/api.c

```diff
@@ -1247,22 +1245,14 @@ MsQuicStreamReceiveComplete(
         !Connection->State.HandleClosed);
 
     if (!Stream->Flags.Started || !Stream->Flags.ReceiveCallPending) {
-        Status = QUIC_STATUS_INVALID_STATE;
         goto Exit;
     }
 
-    Oper = QuicOperationAlloc(Connection->Worker, QUIC_OPER_TYPE_API_CALL);
+    Oper = InterlockedFetchAndClearPointer((void**)&Stream->ReceiveCompleteOperation);
     if (Oper == NULL) {
-        Status = QUIC_STATUS_OUT_OF_MEMORY;
-        QuicTraceEvent(
-            AllocFailure,
-            "Allocation of '%s' failed. (%llu bytes)",
-            "STRM_RECV_COMPLETE operation",
-            0);
-        goto Exit;
+        goto Exit; // Duplicate calls to receive complete
```

Comment:
I feel like this should also be a log point. Since it's kind of bad if an application calls receive complete on a stream twice.